leapsectz/leapsectz.go

```diff
@@ -20,54 +20,109 @@ limitations under the License.
 package leapsectz
 
 import (
+	"bytes"
 	"encoding/binary"
 	"errors"
 	"io"
+	"io/ioutil"
 	"os"
 	"time"
 )
 
 const file = "/usr/share/zoneinfo/right/UTC"
 
 var errBadData = errors.New("malformed time zone information")
+var errBadVersion = errors.New("version is not supported in file")
 
 // LeapSecond represents a leap second
 type LeapSecond struct {
 	Tleap uint32
-	Nleap uint32
+	Nleap int32
+}
+
+// LeapSecond represents a leap second
+type LeapSecondV2 struct {
+	Tleap uint64
+	Nleap int32
+}
+
+// File header struct
```

Comment:
You need to comment starting with the method/struct name to generate godoc.
So `Header is a blah`